### ISSUE TITLE
Introduce box-sizing control

### ISSUE THREAD

**Issue by aristath** [WordPress/gutenberg#52361]
## What problem does this address?
Allowing users to choose between `content-box` and `border-box` will open the door to more customization and design options for blocks (see https://developer.mozilla.org/en-US/docs/Web/CSS/box-sizing)
Sometimes we want the padding, margins and borders to be inside the element, and other times design dictates o have these outside the element. Adding the option will allow users to create more innovative designs.

## What is your proposed solution?
I propose we add a new option in the dimensions section, to allow choosing between content-box and border-box.

**Comment by skorasaurus** [comment]
What about a way to do this through hooks and filters?

**Comment by samanthaberkley** [comment]
Yeah! Just noting that as a block developer, literally in every block we need to set box-sizing. And when we create block style variations, sometimes the canvas needs to be content-box, where it was previously expected to be border-box. And the way how Core blocks have hardcoded border-box is not good for websites that we want to be entirely content-box. Imagine the work.
So there is the need to adjust this in great volumes, in a way that Theme.json would be good and appropriate for this setting.
Resorting to PHP may also be good, but not sufficient.
And if this control is to be available in the UI too, I think that the ToolsPanel is more than sufficient to handle this (or any other control). Because if this is an "advanced" setting an such. Ok it doesn't need to be visible by default in the Dimensions panel. It is going to be appreciated if it is an option in the dropdown menu, to be able to apply it when needed in the UI for our custom blocks and for core blocks too. Although less frequent, this also happens sometimes when we are heavy in development, for more artistic websites that have a lot of absolute positions and animations, in this scenario we may see ourselves in need to change box-sizing on post-development/production together with the client to fix a few things. And having the possibility to see this change without refreshing the page is going to be good. Truth be said, this actually happens more with z-index for each innerblock. If only a z-index control could also be introduced, I would be one step closer to heaven. 😂
Thanks @aristath

**Comment by ghost** [comment]
As a block developer, sometimes we need to change the box-sizing of all Core blocks. Artistic sites and others that have animations, often require to change box-sizing per-case. Also we create block styles variations, and those may have different settings per-canvas. Imagine the work. The volume of content is big, therefore theme.json seems appropriate and easy to adjust those settings.
One can argue that this setting is uncommon for most users. I would agree, and say: that setting is going to be well localized inside the ToolsPanel for those users that need it.
Thank you aristath.
Also z-index control would be good, to manage InnerBlocks z-index.
